go-packages/meep-logger/logger.go

```diff
@@ -17,59 +17,117 @@
 package logger
 
 import (
-	log "github.com/sirupsen/logrus"
+	"net/http"
+	"time"
+	"runtime"
+	"path"
+	"fmt"
+	logrus "github.com/sirupsen/logrus"
 )
 
 var componentName string
 
 type Fields map[string]interface{}
 
 func MeepTextLogInit(name string) {
-	log.SetFormatter(&log.TextFormatter{})
-	log.SetLevel(log.DebugLevel)
+	//logrus.SetLevel(logrus.TraceLevel)
+        logrus.SetLevel(logrus.DebugLevel)
 	componentName = name
 }
 
 func MeepJSONLogInit(name string) {
-	log.SetFormatter(&log.JSONFormatter{})
-	log.SetLevel(log.DebugLevel)
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	//logrus.SetLevel(logrus.TraceLevel)
+        logrus.SetLevel(logrus.DebugLevel)
 	componentName = name
 }
 
+// getLogCaller
+// setReportCaller cannot ignore caller levels, so cannot work in a wrapper. Fix not merged in github, so doing it manually
+func getLogCaller() string {
+	_, file, line, _ := runtime.Caller(2)
+
+        location := fmt.Sprintf("%v:%v", path.Base(file), line)
+	return location
+}
+
```

Comment:
Why not use the built-in logrus method to enable file logging?
https://godoc.org/github.com/sirupsen/logrus#SetReportCaller

Also, you can format the output as you wish using the CallerPrettyfier:
https://github.com/sirupsen/logrus/issues/63

Reply:
this method is good but only comes back one level. As per comment, the reason why it was done so is because since we use a wrapper, we need to go up 2 calls in the stack.. otherwise the log will not be of the caller of our wrapper but from our wrapper itself.. There is a need to call caller(2) in this case as describe in: https://github.com/sirupsen/logrus/issues/972

Reply:
The setReportCaller only goes up one level in the stack. As per the comment at the beginning of the function, since we are in a wrapper, we need to go up 2 level in the stack.. which is why the setReportCaller couldn't be used. If used, it would show the line and file of our wrapper as per https://github.com/sirupsen/logrus/issues/972